pkg/apiserver/apiserver.go

```diff
@@ -102,18 +105,50 @@ func NewAPIServer(clientConfig *rest.Config) (*APIServer, error) {
 		return nil, err
 	}
 
-	return &APIServer{server, make(chan struct{})}, nil
+	return &APIServer{server, make(chan struct{}), aggClient}, nil
 }
 
 // Spawns go routine that exits when apiserver is stopped
-func (as *APIServer) Run() {
+func (as *APIServer) Run() error {
+	const (
+		startupTimeout = 30 * time.Second
+	)
 	go as.server.PrepareRun().Run(as.stopCh)
+
+	// wait for apiservice to have healthy status
+	for t := time.Now().Add(startupTimeout); time.Now().Before(t); {
```

Comment:
call me a n00b, but this loop read confusingly. can we pull out t out of it on its own line?